### ISSUE TITLE
Could NMT work with TF1.2? Model parallel softmax 

### ISSUE THREAD

**Issue by willzyz** [tensorflow/nmt#263]
Hi Thang / NMT authors, 

First question, I wonder if it is at all feasible to have NMT code-base work with TensorFlow 1.2, and if so what is your estimate, and your suggestion of API changes, that need to be made on the code? 

In the seq2seq paper, the softmax is parallelized with multiple GPUs. I wonder first where is the softmax assigned (device/gpu:0, 1, 2), and if we could parallelize the softmax computation across GPUs is there a way to do it? 

Many thanks.

**Comment by oahziur** [comment]
@willzyz There is a [tf-1.2 branch](https://github.com/tensorflow/nmt/tree/tf-1.2), you should be able to checkout the branch and run it on tensorflow 1.2.

By default, the softmax layer is placed on the same gpu as the last RNN layer, so we can minimize memory copy among GPUs. We don't use multiple GPUs for the softmax layer in this codebase.

For parallelized softmax, do you want to parallel the `matmul()` in the softmax layer? If so, I think you can simply split the softmax matrix by column, do the `matmul()` on multiple GPU, and then concat the logits to compute the final probability.
